sdk/messaging/azservicebus/receiver.go

```diff
@@ -358,15 +493,25 @@ func (e *entity) String() (string, error) {
 		return "", errors.New("a queue or subscription was not specified")
 	}
 
-	if e.Subqueue == SubQueueDeadLetter {
+	if e.subqueue == SubQueueDeadLetter {
 		entityPath += "/$DeadLetterQueue"
-	} else if e.Subqueue == SubQueueTransfer {
+	} else if e.subqueue == SubQueueTransfer {
 		entityPath += "/$Transfer/$DeadLetterQueue"
 	}
 
 	return entityPath, nil
 }
 
+func (e *entity) SetSubQueue(subQueue SubQueue) error {
+	if subQueue == SubQueueDeadLetter || subQueue == SubQueueTransfer {
```

Comment:
Centralize?

Reply:
I think you caught this in between reviews.

This is where I centralized it. The two ReceiverWithSubQueue and ProcessorWithSubQueue methods just call into here now. So there's not 3, just 1.